tests/test_aggregates.py

```diff
@@ -17,68 +19,108 @@
+        self.committee = factories.CommitteeHistoryFactory(
+            name='Ritchie for America',
+            cycle=2012,
+        )
+        self.candidate = factories.CandidateDetailFactory(
+            candidate_id='P123',
+            name='Robert Ritchie',
+            election_years=[2012],
+            office='P',
+        )
+        self.candidate_history = factories.CandidateHistoryFactory(
+            candidate_id='P123',
+            name='Robert Ritchie',
+            election_years=[2012],
+            two_year_period=2012,
+            office='P',
+        )
+
+    def make_aggregates(self, factory):
+        return [
+            factory(
+                candidate_id=self.candidate.candidate_id,
+                committee_id=self.committee.committee_id,
+                cycle=self.committee.cycle,
+                total=100,
+                count=5,
             ),
-            (
-                factories.ElectioneeringByCandidateFactory,
-                ElectioneeringByCandidateView,
-                schemas.ElectioneeringByCandidateSchema,
+            factory(
+                candidate_id=self.candidate.candidate_id,
+                committee_id=self.committee.committee_id,
+                cycle=self.committee.cycle - 2,
+                total=100,
+                count=5,
             ),
         ]
-        for factory, resource, schema in params:
-            aggregate = factory(
-                committee_id=self.committee.committee_id,
-                cycle=self.committee.cycle,
+
+    def test_candidate_aggregates_by_committee(self):
+        for factory, resource, schema in self.cases:
+            aggregates = self.make_aggregates(factory)
+            results = self._results(
+                api.url_for(
+                    resource,
+                    committee_id=self.committee.committee_id,
+                    cycle=2012,
+                )
             )
+            assert len(results) == 1
+            serialized = schema().dump(aggregates[0]).data
+            serialized.update({
+                'committee_name': self.committee.name,
+                'candidate_name': self.candidate.name,
+            })
+            assert results[0] == serialized
+
+    def test_candidate_aggregates_by_committee_full(self):
+        for factory, resource, schema in self.cases:
+            aggregates = self.make_aggregates(factory)
             results = self._results(
                 api.url_for(
                     resource,
+                    candidate_id=self.candidate.candidate_id,
                     committee_id=self.committee.committee_id,
                     cycle=2012,
+                    election_full='true',
                 )
             )
-            self.assertEqual(len(results), 1)
-            self.assertEqual(results[0], schema().dump(aggregate).data)
+            assert len(results) == 1
+            serialized = schema().dump(aggregates[0]).data
```

Comment:
These tests might benefit from some comments. I _think_ I've got what's going on here, but a little description would help minimize the head-scratching.